pkg/providers/tinkerbell/comparator.go

```diff
@@ -0,0 +1,51 @@
+package tinkerbell
+
+import (
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
+)
+
+// VersionComparator enables kubernetes version retrieval.
+type VersionComparator interface {
+	GetVersion() string
+}
+
+// ReplicaComparator enables replica number retrieval.
+type ReplicaComparator interface {
+	GetReplicas() int
+}
+
+// KubeadmControlPlaneComparator wraps the KubeadmControlPlane to enable Comparator functions.
+type KubeadmControlPlaneComparator struct {
+	*controlplanev1.KubeadmControlPlane
+}
+
+// GetVersion retrieves KubeadmControlPlane replicas.
+func (k *KubeadmControlPlaneComparator) GetVersion() string {
+	return k.Spec.Version
+}
+
+// GetReplicas retrieves KubeadmControlPlane replicas.
+func (k *KubeadmControlPlaneComparator) GetReplicas() int {
+	return int(*k.Spec.Replicas)
+}
+
+// MachineDeploymentComparator wraps the MachineDeployment to enable Comparator functions.
```

Comment:
it only enable one
maybe something like
```suggestion
// MachineDeploymentComparator implements ReplicaComparator by decorating a MachineDeployment.
```